src/main/kotlin/no/nav/familie/tilbake/behandlingskontroll/BehandlingskontrollService.kt

```diff
@@ -0,0 +1,128 @@
+package no.nav.familie.tilbake.behandlingskontroll
+
+import no.nav.familie.kontrakter.felles.tilbakekreving.Tilbakekrevingsvalg
+import no.nav.familie.tilbake.behandling.BehandlingRepository
+import no.nav.familie.tilbake.behandling.domain.Behandling
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingssteg
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingsstegstatus
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingsstegstatus.AVBRUTT
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingsstegstatus.KLAR
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingsstegstatus.STARTET
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingsstegstatus.VENTER
+import no.nav.familie.tilbake.behandlingskontroll.domain.Behandlingsstegstilstand
+import no.nav.familie.tilbake.common.exceptionhandler.Feil
+import no.nav.familie.tilbake.common.repository.findByIdOrThrow
+import no.nav.familie.tilbake.kravgrunnlag.KravgrunnlagRepository
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+@Service
+class BehandlingskontrollService(private val behandlingsstegstilstandRepository: BehandlingsstegstilstandRepository,
+                                 private val behandlingRepository: BehandlingRepository,
+                                 private val kravgrunnlagRepository: KravgrunnlagRepository) {
+
+    private val log = LoggerFactory.getLogger(this::class.java)
+
+    @Transactional
+    fun fortsettBehandling(behandlingId: UUID) {
+        val behandling = behandlingRepository.findByIdOrThrow(behandlingId)
+        if (behandling.erAvsluttet()) {
+            return
+        }
+        val behandlingsstegstilstand = behandlingsstegstilstandRepository.findByBehandlingId(behandling.id)
+        val aktivStegstilstand = finnAktivStegstilstand(behandlingsstegstilstand)
+
+        if (aktivStegstilstand == null) {
+            val nesteStegMetaData = finnNesteBehandlingsstegMedStatus(behandling, behandlingsstegstilstand)
+            when {
+                behandlingsstegstilstandRepository.findByBehandlingIdAndBehandlingssteg(
+                        behandlingId, nesteStegMetaData.behandlingssteg) == null -> {
```

Comment:
Denne ble litt vanskelig å lese pga. linjedelingen.  Tror kanskje behandlingsstegstilstandRepository.findByBehandlingIdAndBehandlingssteg(behandlingId, nesteStegMetaData.behandlingssteg), kan tilordnes en variabel for lesbarhet.